### PR TITLE
9.0-theme_kit added function to define less styles

### PR DESCRIPTION
**Pull request by GabbasovDinar**

### CODE REVIEW

File: theme_kit/models/theme.py [hunk 1]
```diff
@@ -1,26 +1,312 @@
...
+    top_panel_active_item_bg = fields.Char('Active item Background color', help="Active item Background color for Top Panel")
+    top_panel_active_item_bg_active = fields.Boolean(default=False, help="Menu Bar color for Top Panel")
+
+    top_panel_hover_item_font = fields.Char('Hover item Font color', help="Hover item Font color for Top Panel")
+    top_panel_hover_item_font_active = fields.Boolean(default=False, help="Menu Bar color for Top Panel")
+
+    top_panel_hover_item_bg = fields.Char('Hover item Background color', help="Hover item Background color for Top Panel")
+    top_panel_hover_item_bg_active = fields.Boolean(default=False, help="Menu Bar color for Top Panel")
+
     code = fields.Text('Code', help='technical computed field', compute='_compute_code')
```
Comment (yelizariev) [inline theme_kit/models/theme.py]: rename to ``less``

File: theme_kit/models/theme.py [hunk 2]
```diff
@@ -1,26 +1,312 @@
...
+    left_panel_active_item_bg_active = fields.Boolean(default=False, help="Menu Bar color for Top Panel")
+
+    left_panel_hover_item_font = fields.Char('Hover item Font color', help="Hover item Font color for Left Menu Bar")
+    left_panel_hover_item_font_active = fields.Boolean(default=False, help="Menu Bar color for Top Panel")
 
-            r.code = code
+    left_panel_hover_item_bg = fields.Char('Hover item Background color', help="Hover item Background color for Left Menu Bar")
+    left_panel_hover_item_bg_active = fields.Boolean(default=False, help="Menu Bar color for Top Panel")
+
+    code = fields.Text('Code', help='technical computed field', compute='_compute_code')
```
Comment (yelizariev) [inline theme_kit/models/theme.py]: rename to ``less``

File: theme_kit/models/theme.py [hunk 3]
```diff
@@ -1,26 +1,312 @@
...
+    content_form = fields.Char('Background form color')
+    content_form_active = fields.Boolean(default=False, help="Menu Bar color for Top Panel")
+
+    content_form_text = fields.Char('Text form color')
+    content_form_text_active = fields.Boolean(default=False, help="Menu Bar color for Top Panel")
+
+    content_form_link = fields.Char('Link form color')
+    content_form_link_active = fields.Boolean(default=False, help="Menu Bar color for Top Panel")
+
+    code = fields.Text('Code', help='technical computed field', compute='_compute_code')
```
Comment (yelizariev) [inline theme_kit/models/theme.py]: rename to ``less``

File: theme_kit/models/theme.py [hunk 4]
```diff
@@ -1,26 +1,312 @@
...
 
 
 class Theme(models.Model):
     _name = "theme_kit.theme"
+    name = fields.Char('Name')
+    top_panel_id = fields.Many2one('theme_kit.top_panel', string="Color Schemes for Top Panel")
+    left_panel_id = fields.Many2one('theme_kit.left_panel', string="Color Schemes for Left Panel")
+    content_id = fields.Many2one('theme_kit.content', string="Color Schemes for Content")
+
+    code = fields.Text('Code', help='technical computed field', compute='_compute_code')
```
Comment (yelizariev) [inline theme_kit/models/theme.py]: ``code`` here is OK

File: theme_kit/models/theme.py [hunk 5]
```diff
@@ -1,26 +1,312 @@
...
+            if r.left_panel_id:
+                code = code + r.left_panel_id.code
+            if r.content_id:
+                code = code + r.content_id.code
+            bundle = AssetsBundle('theme_kit.dummy')
+            assets = LessStylesheetAsset(bundle, inline=code, url='')
+            cmd = assets.get_command()
+            source = assets.get_source()
+            # source = '\n'.join([asset.get_source() for asset in assets])
+            compiled = bundle.compile_css(cmd, source)
```
Comment (yelizariev) [inline theme_kit/models/theme.py]: move these lines to new function _less2css

```

+            bundle = AssetsBundle('theme_kit.dummy')
 +            assets = LessStylesheetAsset(bundle, inline=code, url='')
 +            cmd = assets.get_command()
 +            source = assets.get_source()
 +            # source = '\n'.join([asset.get_source() for asset in assets])
 +            compiled = bundle.compile_css(cmd, source)
```

File: theme_kit/models/theme.py [hunk 6]
```diff
@@ -1,26 +1,312 @@
...
+            compiled = '''<style type="text/css" id="theme_kit.custom_css">''' + compiled + '''</style>'''
```
Comment (yelizariev) [inline theme_kit/models/theme.py]: <style type="text/css" id="custom_css"

Comment (yelizariev) [inline theme_kit/models/theme.py]: i.e. no need for ``theme_kit.`` prefix

**Comment by yelizariev** [comment]
Fix warnings

```
2016-10-04 09:46:39	WARNING	server	openerp.modules.loading:348 load_modules The model theme_kit.top_panel has no access rules, consider adding one. E.g. access_theme_kit_top_panel,access_theme_kit_top_panel,model_theme_kit_top_panel,,1,0,0,0
2016-10-04 09:46:39	WARNING	server	openerp.modules.loading:348 load_modules The model theme_kit.theme has no access rules, consider adding one. E.g. access_theme_kit_theme,access_theme_kit_theme,model_theme_kit_theme,,1,0,0,0
2016-10-04 09:46:39	WARNING	server	openerp.modules.loading:348 load_modules The model theme_kit.left_panel has no access rules, consider adding one. E.g. access_theme_kit_left_panel,access_theme_kit_left_panel,model_theme_kit_left_panel,,1,0,0,0
```

**Comment by yelizariev** [comment]
lint errors:
```

theme_kit/models/theme.py:120:1: E303 too many blank lines (3)
theme_kit/models/theme.py:422:13: E303 too many blank lines (2)
theme_kit/data/theme_data.xml:1: [W7908(missing-newline-extrafiles), ]  Missing newline

```

**Comment by yelizariev** [comment]
1475647378340screensave
1475647380494screensave

File: theme_kit/models/theme.py [hunk 7]
```diff
@@ -22,9 +24,19 @@ def _compute_code(self):
...
+                js_code = r.custom_js
+                js_code = 'try {' + js_code + '''
+                    } catch (err) {
+                      console.log('Error' + err.name + ":" + err.message + ". " + err.stack);
+                      alert('Error' + err.name + ":" + err.message + ". " + err.stack);
+                    }'''
+                code = code + '''<script type="text/javascript" id="custom_js">''' + js_code + '''</script>'''
+                r.code = code
             else:
                 r.code = code
```
Comment (yelizariev) [inline theme_kit/models/theme.py]: r.code = code
            else:
                r.code = code

can be simplified:

            r.code = code

File: theme_kit/models/theme.py [hunk 8]
```diff
@@ -9,6 +9,8 @@ class Theme(models.Model):
     top_panel_id = fields.Many2one('theme_kit.top_panel', string="Color Schemes for Top Panel")
     left_panel_id = fields.Many2one('theme_kit.left_panel', string="Color Schemes for Left Panel")
     content_id = fields.Many2one('theme_kit.content', string="Color Schemes for Content")
+    custom_css = fields.Text(string="Custom CSS/LESS", default=False)
+    custom_js = fields.Text(string="Custom JS", default=False)
```
Comment (yelizariev) [inline theme_kit/models/theme.py]: Add example of usage to index.rst, e.g.
css
```

.test-ribbon{
    width: 200px;
    top: 25px;
    left: -50px;
    text-align: center;
    line-height: 50px;
    letter-spacing: 1px;
    color: #f0f0f0;
    -webkit-transform: rotate(-45deg);
    -ms-transform: rotate(-45deg);
    -moz-transform: rotate(-45deg);
    -o-transform: rotate(-45deg);
    transform: rotate(-45deg);
    z-index: 9999;
    position: fixed;
    box-shadow: 0 0 3px rgba(0,0,0,.3);
    background: rgba(255,0,0,.6);
    pointer-events: none;
}

.test-ribbon b {
    font-size: 20px;
}
```

js
```

$(document).ready(function(){
    var ribbon = $('<div class="test-ribbon"/>');
    $('body').append(ribbon);
    ribbon.html("Demo");
})
```

**Comment by GabbasovDinar** [comment]
@yelizariev DONE

**Pull request merged**